raimitigations/dataprocessing/imputer/iterative_imputer.py

```diff
@@ -0,0 +1,292 @@
+        """
+        param_err = type(self.iterative_params) != dict
+
+        if param_err:
+            raise ValueError("ERROR: 'iterative_params' is not a dict. Check the documentation for more information.")
+
+        keys = self.iterative_params.keys()
+        sklearn_params = [
+            "estimator",
+            "missing_values",
+            "sample_posterior",
+            "max_iter",
+            "tol",
+            "n_nearest_features",
+            "initial_strategy",
+            "imputation_order",
+            "skip_complete",
+            "min_value",
+            "max_value",
+            "random_state",
+        ]
+        for sklearn_param in sklearn_params:
+            if sklearn_param not in keys:
+                raise ValueError(
+                    "ERROR: expected the key "
+                    + f"{sklearn_param}"
+                    + " in the dictionary:\n"
+                    + f"iterative_params: {self.iterative_params}"
+                )
+
+    # -----------------------------------
+    def _fit(self):
+        """
+        Fit method complement used specifically for the current class.
+        The following steps are executed: (i) check if 'self.iterative_params'
+        dictionary is properly set, (ii) check for categorical columns
+        (iii) exclude categorical columns if 'enable_encoder' is false,
+        otherwise, apply ordinal encoder, (iv) create and fit the
+        IterativeImputer object over df.
+        """
+        self._check_valid_dict()
+
+        all_cat_cols = get_cat_cols(self.df)
+        all_num_cols = [value for value in list(self.df) if value not in all_cat_cols]
+
+        df_valid = pd.DataFrame()
+        self.valid_cols = []
+
+        if not isinstance(self.enable_encoder, bool):
+            raise ValueError("ERROR: 'enable_encoder' is a boolean parameter, use True/False.")
+
+        elif self.enable_encoder is False:
+            warnings.warn(
+                "\nWARNING: Categorical columns will be excluded from the iterative imputation process.\n"
+                + "If you'd like to include these columns, you need to set 'enable_encoder'=True.\n"
+                + "If you'd like to use a different type of encoding before imputation, consider using the Pipeline class and call your own encoder before calling this subclass for imputation.",
+            )
+            df_valid = self._get_df_subset(self.df, all_num_cols)
+
+        else:
+            warnings.warn(
+                "\nWARNING: 'enable_encoder'=True and categorical columns will be encoded using ordinal encoding before applying the iterative imputation process.\n"
+                + "If you'd like to use a different type of encoding before imputation, consider using the Pipeline class and call your own encoder before calling this subclass for imputation.",
+            )
+            ordinal_encoder = EncoderOrdinal(df=self.df, col_encode=all_cat_cols, unknown_value=np.nan)
```

Comment:
Yes OHE might be more suited for some estimators, however I didn't use it given the imputation of OHE would certainly be very odd. If you try to impute multiple columns that represent a single feature, because the way the imputer works it wouldn't understand this is a single feature and you could end up with a 1 in multiple columns with the feature having multiple values per instance.
As for choosing an encoding suitable to a certain estimator, that's where the flexibility the user has of choosing the estimator comes in handy as well as the printed messages that instruct the user to use a different encoding using the Pipeline class if they see fit.

Reply:
Makes sense